### PR TITLE
fix docs for narrow_copy

### PR DESCRIPTION
**Pull request by weiyangfb**
- fix docs for narrow_copy and make it appear on doc page

### CODE REVIEW

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -1457,11 +1457,11 @@ def callable(a, b) -> number
                r"""
 narrow_copy(dimension, start, length) -> Tensor
 
-Same as :meth:`Tensor.narrow` except returning a copy rather
-than shared storage.  This is primarily for sparse tensors, which
-do not have a shared-storage narrow method.  Calling ```narrow_copy``
-with ```dimemsion > self._sparseDims()``` will return a copy with the
-relevant dense dimension narrowed, and ```self.shape``` updated accordingly.
+Same as :func:`torch.narrow` except that :meth:`~FloatTensor.narrow_copy` is a Tensor method
```
Comment (SsnL) [inline torch/_tensor_docs.py]: Does `~torch.Tensor.narrow_copy` not work?

File: torch/_tensor_docs.py [hunk 2]
```diff
@@ -1457,11 +1457,11 @@ def callable(a, b) -> number
...
+and it returns a copy rather than shared storage. This is primarily for sparse tensors, which
+do not have a shared-storage narrow method. Calling :meth:`~FloatTensor.narrow_copy`
+with ``dimemsion > self._sparseDims()`` will return a copy with the
```
Comment (SsnL) [inline torch/_tensor_docs.py]: So.. if `dimemsion <= self._sparseDims()` it actually returns a shared storage tensor?

File: torch/_tensor_docs.py (hunk 1 above)
Comment (weiyangfb) [reply torch/_tensor_docs.py]: just tried, but no, it doesn't work

File: torch/_tensor_docs.py (hunk 2 above)
Comment (SsnL) [reply torch/_tensor_docs.py]: Then this is misleading. Can we just delete this last sentence?

Comment (weiyangfb) [reply torch/_tensor_docs.py]: here it says 'return a copy', maybe it is ok to keep it? I think it is helpful to mention the behavior of narrowing and copy dense tensor somewhere.

File: torch/_tensor_docs.py [hunk 3]
```diff
@@ -1459,9 +1459,8 @@ def callable(a, b) -> number
 
 Same as :func:`torch.narrow` except that :meth:`~FloatTensor.narrow_copy` is a Tensor method
 and it returns a copy rather than shared storage. This is primarily for sparse tensors, which
-do not have a shared-storage narrow method. Calling :meth:`~FloatTensor.narrow_copy`
-with ``dimemsion > self._sparseDims()`` will return a copy with the
-relevant dense dimension narrowed, and ``self.shape`` updated accordingly.
+do not have a shared-storage narrow method. :meth:`~FloatTensor.narrow_copy` can be applied on
+both of ``sparse_dim`` and ``dense_dim``.
```
Comment (apaszke) [inline torch/_tensor_docs.py]: Why do we even have this method? We really should avoid duplicating all of our implementations just because someone starts using sparse arrays.

Comment (weiyangfb) [reply torch/_tensor_docs.py]: @apaszke I think this is not a duplicate of existing Ops. The closest one to this is `narrow()` which returns a view rather than a new copy.

Comment (SsnL) [reply torch/_tensor_docs.py]: @apaszke because narrow can't work on sparse tensors.

Comment (apaszke) [reply torch/_tensor_docs.py]: Except `narrow_copy` is more like `.narrow(...).clone()`, and introducing ops that can be this easily composed out of other operations we have seems like a bad idea (that's a quadratic increase in the size of a library in case of 2 ops). Finally, we use `copy` to indicate that we move data from one tensor to another, while what you're describing here is more of a `clone`.

@SsnL I understand that narrow can't return a view in the sparse case, but I can't see how adding a specialized op helps here. What will you do if someone asks you to give them `sparse_tensor[2:5]`? We will have to figure out how to handle those ambiguities in a single API anyway, so adding this function is unnecessary.

Comment (SsnL) [reply torch/_tensor_docs.py]: I probably didn't make myself clear. Let me try again.

Since it is impossible to do a `narrow`-like operation on a sparse tensor (returning a view), we shouldn't support `sparse_tensor.narrow(...)` or `sparse_tensor[2:5]`. If we were to make `sparse_tensor[2:5]` return a copy, then it would be inconsistent and confusing because people would expect `sparse_tensor[2:5] = x` to behave similarly as `dense_tensor[2:5] = x`. However, a `narrow`-like operation is very useful, even if it doesn't return a view, but it is not easily implementable using existing ops. Therefore, I really think that we should make something obviously different from `narrow` API, either with a new name or with an additional keyword.

Comment (apaszke) [reply torch/_tensor_docs.py]: FWIW `sparse_tensor[2:5] = x` can still be made to work with sparse tensors, even though `sparse_tensor[2:5]` does not return a view. This calls `__setitem__`, so you can do both indexing and assignment in one step. I'm just not convinced that duplicating every single item in our API is a good idea. What if someone later comes with another sparsity pattern that can represent a different subset of those operations cheaply? Do we start adding more of them? Forbidding regular indexing syntax on sparse tensors will make them almost unusable, and I don't think that's the path we want to take.

Comment (weiyangfb) [reply torch/_tensor_docs.py]: @apaszke I took a quick look at the API docs, and found 3 ops (there might be more that are not documented, but likely there should be < 10 ops) have returned tensors share storage with inputs. It is possible that in the future we will need to implement them for sparse tensor under different names, but still it is bounded by 10. So maybe we can afford to have these special ops for sparse?

Also, correct me if I am wrong, but I think it might be tricky to allow `sparse_tensor[2:5] = x`, the reason is `x` might have different indices and values compare to `sparse_tensor[2:5]`, and so a new copy of `sparse_tensor` is needed.

> Forbidding regular indexing syntax on sparse tensors will make them almost unusable, and I don't think that's the path we want to take.
I think this largely depends on user cases. At least for now, to be able to do `sum()`, `nn.Linear()`, `softmax()`, etc can already be useful for some particular models. Ideally the purpose of sparse tensor is not that it is super generalizable so that any models can be implemented in sparse, but is that it can make some applications possible or maybe more efficient.
